cmd/mdatagen/internal/templates/metrics.go.tmpl

```diff
@@ -396,6 +465,7 @@ func (mb *MetricsBuilder) Record{{ $name.Render }}DataPoint(ts pcommon.Timestamp
 		return fmt.Errorf("failed to parse {{ $metric.Data.MetricValueType.BasicType }} for {{ $name.Render }}, value was %s: %w", inputVal, err)
 	}
 	{{- end }}
+  if !mb.config.enabledSetByUser {
```

Comment:
We shouldn't care whether the `attributes` set by user or not. Some attributes might by already disabled by default so we have to do the re-aggregation regardless

Reply:
heard, originally this was meant to create a situation that only calls the reagg function when needed but since we're not going with that approach it is indeed not needed.